client/components/app.jsx

```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
 import DisplayPic from './display.jsx';
 import LeftArrow from './leftArrow.jsx';
 import RightArrow from './rightArrow.jsx';
+import HeartButton from './heartButton.jsx';
 import Modal from './modal.jsx';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      itemName: 'BTS BT21 Official Pyjamas Set',
-      images: [
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_1.jpg',
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_2.jpg',
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_3.jpg'
-      ],
+      images: [],
       currIndex: 0,
       translateVal: 0,
-      show: false
+      show: false,
+      like: false,
+      productId: null
     };
     this.nextPicture = this.nextPicture.bind(this);
     this.prevPicture = this.prevPicture.bind(this);
     this.selectedPic = this.selectedPic.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
+    this.toggleHeart = this.toggleHeart.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+
   }
+
   toggleModal (event) {
     this.setState({ show: !this.state.show})
   }
 
+  async toggleHeart (event) {
+    await this.setState({ like: !this.state.like});
+    axios.put('/products', {
+      productId: this.state.productId,
+      like: this.state.like
+    })
+      .then(response => {
+        console.log(response, 'hello');
+      })
+      .catch(error => {
+        console.log(err);
+      });
+  }
+
+  componentDidMount() {
+    var randomNum = (max) => {
+      return Math.floor(Math.random() * max) + 1;
+    };
+    var productId = randomNum(5);
```

Comment:
You should discuss w/ your team how each module is determining for which productId you are rendering information.